src/main.c

```diff
@@ -1052,7 +1052,22 @@ static void brake(data *d) {
     }
 
     VESC_IF->timeout_reset();
-    VESC_IF->mc_set_brake_current(d->float_conf.brake_current);
+
+    // If brake current is set to 0 don't do anything
+    if (d->float_conf.brake_current == 0) {
+        return;
+    }
+
+    // Use brake current over certain ERPM to prevent the board skidding to a stop when deactivated
+    // at speed?
+    if (d->motor.abs_erpm > 2000) {
+        VESC_IF->mc_set_brake_current(d->float_conf.brake_current);
+        return;
+    }
+
+    // Use DC control mode as it has better holding power
+    // Also improves with 6.05 shorting feature
+    VESC_IF->mc_set_duty(0);
```

Comment:
I wonder if you'd always want to do this, not just when movement was detected (on line 1046 above) and it didn't time out?

Reply:
> I wonder if you'd always want to do this, not just when movement was detected (on line 1046 above) and it didn't time out?

that would be a question for the VESC Discord. My guess is that it would waste energy and drain our batteries. That's why this entire logic exists (we used to not have it and brake current was keeping our ESCs warm...)

Reply:
No response on the discord, I guess I'll do some testing when I have a chance...